src/safeds/data/tabular/plotting/_table_plotter.py

```diff
@@ -104,6 +104,97 @@ def box_plots(self) -> Image:
         fig.tight_layout()
         return _figure_to_image(fig)
 
+    def violin_plots(self, theme: Literal["dark", "light"] = "light") -> Image:
+        """
+        Create a violin plot for every numerical column.
+
+        Returns
+        -------
+        plot:
+            The violin plot(s) as an image.
+
+        Raises
+        ------
+        NonNumericColumnError
+            If the table contains only non-numerical columns.
+
+        Examples
+        --------
+        >>> from safeds.data.tabular.containers import Table
+        >>> table = Table({"a":[1, 2], "b": [3, 42]})
+        >>> image = table.plot.violin_plots()
+        """
+        numerical_table = self._table.remove_non_numeric_columns()
+        if numerical_table.column_count == 0:
+            raise NonNumericColumnError("This table contains only non-numerical columns.")
+        from math import ceil, nan
+
+        import matplotlib.pyplot as plt
+
+        style = "dark_background" if theme == "dark" else "default"
+        with plt.style.context(style):
+            if theme == "dark":
+                plt.rcParams.update(
+                    {
+                        "text.color": "white",
+                        "axes.labelcolor": "white",
+                        "axes.edgecolor": "white",
+                        "xtick.color": "white",
+                        "ytick.color": "white",
+                        "grid.color": "gray",
+                        "grid.linewidth": 0.5,
+                    },
+                )
+            else:
+                plt.rcParams.update(
+                    {
+                        "grid.linewidth": 0.5,
+                    },
+                )
+
+            columns = numerical_table.to_columns()
+            columns = [column._series.drop_nulls() for column in columns]
+            max_width = 3
+            number_of_columns = len(columns) if len(columns) <= max_width else max_width
+            number_of_rows = ceil(len(columns) / number_of_columns)
+
+            fig, axs = plt.subplots(nrows=number_of_rows, ncols=number_of_columns)
+            line = 0
+            for i, column in enumerate(columns):
+                data = column.to_list()
+                if len(column) == 0:
+                    data = [nan, nan]
```

Comment:
Any way to cover this branch?

Reply:
This branch is already covered by this testcase: test_should_match_dark_snapshot with id: empty.
Violin plots will raise an error if the column is of size 0. Thats why we needed to fill the data with at least two NaN so that it won't raise an error. Instead it will show an empty plot which is better.
Therefore it is covered by the mentioned test as the empty table will trigger this case.

Reply:
There is no test with ID `empty` for `Table.plot.violin_plots`. Maybe you confused that with `Column.plot.violin_plot`?

Reply:
I will look into it

Reply:
Indeed I confused it, but after looking into it I noticed that the lines of code which are uncovered are not necessary. The empty case is already catched and will rise a NonNumericColumnError.